starter/app/src/main/java/com/example/android/politicalpreparedness/representative/RepresentativeFragment.kt

```diff
@@ -65,49 +64,58 @@ class DetailFragment : Fragment() {
         }
 
         binding.buttonLocation.setOnClickListener {
-            getLocation()
+            if (checkLocationPermissions()) {
+                getLocation()
+            }
```

Comment:
If location is not allowed you need to show a message to the user to enable the location. Are you implementing it on the next PR?

Reply:
Done